packages/block-library/src/spacer/edit.js

```diff
@@ -93,9 +93,13 @@ const SpacerEdit = ( {
 		return editorSettings?.disableCustomSpacingSizes;
 	} );
 	const { orientation } = context;
-	const { orientation: parentOrientation, type } = parentLayout || {};
+	const {
+		orientation: parentOrientation,
+		type,
+		default: { type: defaultType } = {},
+	} = parentLayout || {};
 	// Check if the spacer is inside a flex container.
-	const isFlexLayout = type === 'flex';
+	const isFlexLayout = type === 'flex' || defaultType === 'flex';
```

Comment:
Theoretically could a block have a type that is not `flex` but also have a `defaultType` of `flex`?

For example, further down the track, if we were to add both spacers and grid layout to the Gallery block, there could be a situation where `type` is `grid` but the block still has a `defaultType` of `flex` 🤔

If we need to guard against something like that, would the following work?

```suggestion
	const isFlexLayout = type === 'flex' || ( ! type && defaultType === 'flex' );
```

Apologies if I'm overthinking it!

Reply:
Hmm yea that's a good point, it's unlikely but possible that a block sets its default as flex and enables `allowSwitching`. And your suggestion won't do any harm!